crates/gitbutler-repo/tests/hooks.rs

```diff
@@ -0,0 +1,163 @@
+use gitbutler_repo::hooks;
+
+#[cfg(test)]
+mod tests {
+
+    use std::{
+        collections::HashMap,
+        path::{Path, PathBuf},
+    };
+
+    use gitbutler_repo::staging::{filter_diff_by_hunk_ids, HunkId};
+    use gitbutler_testsupport::{Case, Suite};
+    use hooks::{ErrorData, HookResult, MessageData};
+
+    use super::*;
+
+    #[test]
+    fn pre_commit_hook_success() -> anyhow::Result<()> {
```

Comment:
Good. Can we also add a test case asserting that only the "selected" files are involved in the execution?

Reply:
It already does that in the reject case. It's first run with an empty selection and passes, then with the selection and fails.